chain/bitcoind_client.go

```diff
@@ -428,17 +416,116 @@ func (c *BitcoindClient) Stop() {
 	}
 
 	close(c.quit)
-	c.client.Shutdown()
+
+	// Remove this client's reference from the bitcoind connection to
+	// prevent sending notifications to it after it's been stopped.
+	c.chainConn.rescanClientsMtx.Lock()
+	delete(c.chainConn.rescanClients, c.clientID)
+	c.chainConn.rescanClientsMtx.Unlock()
+
 	c.notificationQueue.Stop()
 }
 
 // WaitForShutdown blocks until both the client has finished disconnecting
 // and all handlers have exited.
 func (c *BitcoindClient) WaitForShutdown() {
-	c.client.WaitForShutdown()
 	c.wg.Wait()
 }
 
+// rescanHandler handles the logic needed for the caller to trigger a chain
+// rescan.
+//
+// NOTE: This must be called as a goroutine.
+func (c *BitcoindClient) rescanHandler() {
+	defer c.wg.Done()
+
+	for {
+		select {
+		case update := <-c.rescanUpdate:
+			switch update := update.(type) {
```

Comment:
We're missing the two chain hash slice instances. Atm, these are used for tx confirmation notifications.

Reply:
Fixed.